crates/pop-cli/src/commands/up/contract.rs

```diff
@@ -196,11 +192,20 @@ impl From<UpContractCommand> for UpOpts {
 	}
 }
 
+/// Checks if a contract has been built by verifying the existence of the build directory.
+///
+/// # Arguments
+/// * `path` - An optional path to the project directory. If no path is provided, the current directory is used.
+pub fn has_contract_been_built(path: Option<&Path>) -> bool {
+	path.unwrap_or_else(|| Path::new(".")).join("target/ink").exists()
```

Comment:
Sorry to say, I dont think this is an accurate enough check. If I delete the contract artifacts within target/ink, leaving the directory, it passes this check and then fails with another error. I think it should look for all the required artifacts and if any one of them is missing, trigger a build.

Reply:
You are right, checking only the .contract file as is the one used for the `cargo-contract` crate